Ng ticket 2

nrg51098 opened the pull request:
Description of PR that completes issue here...

## Changes

- Item 1
- Item 2
- Item 3

## Requests / Responses

If this PR contains code that defines a new request/response, or changes an existing one, please put the JSON representations here.

**Request**

POST `/products` Creates a new product

```json
{
    "title": "Kite",
    "product_type_id": 1,
    "description": "Red. It flies high.",
    "quantity": 5
}
```

**Response**

HTTP/1.1 201 OK

```json
{
    "id": 54,
    "title": "Kite",
    "product_type_id": 1,
    "description": "Red. It flies high.",
    "quantity": 5
}
```

## Testing

Description of how to test code...

- [ ] Run migrations
- [ ] Run test suite
- [ ] Seed database


## Related Issues

- Fixes #85
- Fixes #22

nrg51098 committed:
fixed the ticket # 2

nrg51098 committed:
fixed the critical issue with the try catch block

hannahhall commented on bangazonapi/views/profile.py:
```diff
@@ -156,7 +156,7 @@ def cart(self, request):
                 cart = {}
                 cart["order"] = OrderSerializer(open_order, many=False, context={'request': request}).data
-                cart["order"]["line_items"] = line_items.data
+                # cart["order"]["line_items"] = line_items.data
```
Instead of commenting out code that's not being used it's better to delete it

hannahhall commented on bangazonapi/views/product.py:
```diff
@@ -240,38 +242,44 @@ def list(self, request):
+            serializer = ProductSerializer(
+                products, many=True, context={'request': request})
+            return Response(serializer.data)
+        except ZeroDivisionError as ex:
```
This exception doesn't need to be here

hannahhall commented on bangazonapi/views/product.py:
```diff
@@ -152,6 +152,8 @@ def retrieve(self, request, pk=None):
             product = Product.objects.get(pk=pk)
             serializer = ProductSerializer(product, context={'request': request})
             return Response(serializer.data)
+        except ZeroDivisionError as ex:
```
This exception doesn't need to be here

hannahhall commented on bangazonapi/models/product.py:
```diff
@@ -61,9 +61,12 @@ def average_rating(self):
+        try:
+            avg = total_rating / len(ratings)
+            return avg
+        except:
```
Since you have this exception here you don't need the other ones in the other files. Like the other ones though this one should have the `ZeroDivisionError as ex`

nrg51098 commented:
Description of PR that completes issue here...

## Changes

- Fixed the zero division error which was a critical issue in the product models
- Fixed the ticket number 1 to remove the line_items from the response object

## Requests / Responses

If this PR contains code that defines a new request/response, or changes an existing one, please put the JSON representations here.

**Request**

GET `/profile/cart` returns the correct json

**Response**
**BEFORE**
```json
{
    "id": 2,
    "url": "http://localhost:8000/orders/2",
    "created_date": "2019-04-12",
    "payment_type": null,
    "customer": "http://localhost:8000/customers/7",
    "lineitems": [" "],
    "line_items": [" "]
}
```
**AFTER FIX**

```json
{
    "id": 2,
    "url": "http://localhost:8000/orders/2",
    "created_date": "2019-04-12",
    "payment_type": null,
    "customer": "http://localhost:8000/customers/7",
    "lineitems": [" "]
}
```
## Related Issues

- Fixes #2
- Fixes #11
